modules/navier_stokes/src/kernels/INSMomentumTractionFormRZ.C

```diff
@@ -98,9 +86,9 @@ INSMomentumTractionFormRZ::computeQpOffDiagJacobian(unsigned jvar)
   // If we're getting the pressure Jacobian contribution, and we
   // integrated the pressure term by parts, there is an extra term for
   // RZ.
-  if (jvar == _p_var_number && _component == 0 && _integrate_p_by_parts)
+  if (jvar == _p_var_number && _component == _rz_radial_coord && _integrate_p_by_parts)
   {
-    const Real r = _q_point[_qp](0);
+    const auto r = _q_point[_qp](_rz_radial_coord);
```

Comment:
want to use ad_q_point for displaced mesh Jacobian or not?

Reply:
Not really for the hand-coded objects. I've made no effort to support 100% accurate Jacobians for displaced meshes in the hand-coded objects to date. If wanting to add that support, I think it should happen in a larger sweep.

I'm still debating whether just to remove the INS objects in general. However, they have been very helpful in checking for bugs in the INSAD implementation. In an ideal world I would use the INSFE objects for that ... or remove the INSFE objects.

Reply:
Also in that same larger discussion, for non-displaced meshes the INS Jacobians are 100% accurate and so the INS objects have faster run-times than INSAD so that would be a good argument for keeping them around. Anyway this is not on-topic haha

Reply:
ah ok that makes sense. 
INSAD will use ad_q_point and INS should consistently use _q_point and manually do the derivative later if it s ever needed

yeah INSFE vs INS will have to be decided some day